src/process/ProcessManagerImpl.cpp

```diff
@@ -701,13 +726,23 @@ std::weak_ptr<ProcessExitEvent>
 ProcessManagerImpl::runProcess(std::string const& cmdLine, std::string outFile)
 {
     std::lock_guard<std::recursive_mutex> guard(mProcessesMutex);
+    if (!mTmpDir)
+    {
+        mTmpDir =
+            std::make_unique<TmpDir>(mApp.getTmpDirManager().tmpDir("process"));
+    }
+
     auto pe =
         std::shared_ptr<ProcessExitEvent>(new ProcessExitEvent(mIOContext));
 
     std::weak_ptr<ProcessManagerImpl> weakSelf(
         std::static_pointer_cast<ProcessManagerImpl>(shared_from_this()));
+
+    auto tempFile =
+        mTmpDir->getName() + "/temp-" + std::to_string(mTempFileCount++);
```

Comment:
`mTempFileCount` is incremented from inside a lock_guard, so it's safe to use a regular int as opposed to `std::atomic`